gspread/worksheet.py

```diff
@@ -2607,6 +2607,21 @@ def unmerge_cells(self, name):
 
         return self.spreadsheet.batch_update(body)
 
+    def get_notes(self, default_empty_value = None):
+        """Returns a list of lists containing all notes in the sheet, or the empty list if the
+        sheet does not have a note.
```

Comment:
I tried your code it works fine.

Though there is an unexpected detail:

The list built and returned to the user could be smaller than the list of data returned to the user.
The point is, I have notes up to the row 3 and I have data up to the row 10.
The returned list of notes has a length of 3 because after row 3 there are o more notes.

somehow, we should mention that resulting matrix size. I don't have the right comment yet but we'll find something.

please add a note to the documentation, as follow with some phrase like bellow (feel free to change the text if you find something better (mind the spaces they matter in rst documentation format)):

```rst
.. note::

   The resulting matrix is as long as the last row holding a note (could be smaller than the last data row)
   The resulting matrix is as large as the last column holding a note (could be smaller than the last data column)
```

on the top of that please add the documenation markers for the typing of your arguments and returned values.
ask for any help if needed.